### ISSUE TITLE
valid-jsdoc fails with missing return error for interface methods

### ISSUE THREAD

**Issue by germanz** [eslint/eslint#9978]
**Tell us about your environment**

* **ESLint Version: 4.17.0**
* **Node Version: v8.3.0**
* **npm Version:**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
'use strict';

module.exports = {
  extends: 'eslint:recommended',
  parserOptions: {
    ecmaVersion: 6
  },
  env: {
    browser: true,
    es6: true
  },
  rules: {
    'valid-jsdoc': ['error', {
      requireReturn: false,
      requireReturnDescription: false,
      requireParamDescription: false
    }]
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
"use strict";

/**
 * A thing interface.
 * @interface
 */
function Thing() {}

/**
 * An interface method which returns
 * @return {Object}
 */
Thing.prototype.bar = function() {};
```

```bash
$(npm bin)/eslint .
```

**What did you expect to happen?**
No valid-jsdoc error. As according to [JsDoc documentation](http://usejsdoc.org/tags-interface.html):

> You do not need to add the @interface tag to each member of the interface (for example, the interface's instance methods).

**What actually happened? Please include the actual, raw output from ESLint.**
```Unexpected @return tag; function has no return statement.```

**Comment by platinumazure** [comment]
Hi @germanz, apologies for letting this sit so long.

I appreciate the JSDoc spec citation; it has convinced me that this is definitely a bug, and I've labeled this issue as "accepted".

**Comment by jouncealimb** [comment]
I was wondering if I could take a look at this

**Comment by platinumazure** [comment]
@jouncealimb Please feel free, thanks! :smile:

**Comment by nzakas** [comment]
Thanks for the suggestion. As of yesterday, we have decided to officially [end-of-life JSDoc support](https://eslint.org/blog/2018/11/jsdoc-end-of-life) in ESLint. All JSDoc features are now deprecated and we won't be fixing bugs or making any further improvements to those features.

We are recommending that people transition over to use the [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) plugin instead of the built-in rules in ESLint.

Thanks for understanding and we apologize for any inconvenience.
